certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/message/service/HandleMessageDomainService.java

```diff
@@ -17,6 +18,45 @@ public class HandleMessageDomainService {
 
   public Message handle(Message message, boolean handle, Certificate certificate,
       ActionEvaluation actionEvaluation) {
+    if (message.type() == MessageType.COMPLEMENT) {
+      return handleComplementMessage(message, handle, certificate, actionEvaluation);
+    }
+
+    return handleAdministrativeMessage(message, handle, certificate, actionEvaluation);
+  }
+
+  private Message handleAdministrativeMessage(Message message, boolean handle,
+      Certificate certificate,
+      ActionEvaluation actionEvaluation) {
+    if (!certificate.allowTo(
+        CertificateActionType.HANDLE_MESSAGE,
+        Optional.of(actionEvaluation)
+    )) {
+      throw new CertificateActionForbidden(
+          "Not allowed to handle administrative message on certificate for %s"
+              .formatted(certificate.id().id()),
+          certificate.reasonNotAllowed(CertificateActionType.HANDLE_MESSAGE, Optional.empty())
+      );
+    }
+
+    if (handle) {
+      deleteDraftAnswerIfExists(message);
+      message.handle();
+    } else {
+      message.unhandle();
+    }
+
+    return messageRepository.save(message);
+  }
+
+  private static void deleteDraftAnswerIfExists(Message message) {
+    if (message.answer() != null && message.answer().status() == MessageStatus.DRAFT) {
```

Comment:
Just a thought, but for certificates, we are able to just call isDraft() function that returns true if draft, and false if not.

Should implement the same logic in answer?

```java
  public boolean isDraft() {
    return MessageStatus.DRAFT.equals(this.status);
  }
```

Reply:
done